packages/rolldown/src/plugin/index.ts

```diff
@@ -249,9 +178,54 @@ export type SequentialPluginHooks =
   | 'renderChunk'
   | 'transform'
 
-export type AddonHooks = 'banner' | 'footer' | 'intro' | 'outro'
+export type AddonHooks = '' /* 'banner' | 'footer' | 'intro' | 'outro' */
+
+export type OutputPluginHooks =
+  | 'augmentChunkHash'
+  | 'generateBundle'
+  | 'outputOptions'
+  | 'renderChunk'
+  // | 'renderDynamicImport'
+  | 'renderError'
+  | 'renderStart'
+  // | 'resolveFileUrl'
+  // | 'resolveImportMeta'
+  | 'writeBundle'
 
 export type ParallelPluginHooks = Exclude<
   keyof FunctionPluginHooks | AddonHooks,
   FirstPluginHooks | SequentialPluginHooks
 >
+
+export type PluginHooks = {
+  [K in keyof FunctionPluginHooks]: ObjectHook<
+    K extends AsyncPluginHooks
+      ? MakeAsync<FunctionPluginHooks[K]>
+      : FunctionPluginHooks[K]
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    // TODO
+    // K extends ParallelPluginHooks ? { sequential?: boolean } : {}
+  >
+}
+
+export type AddonHookFunction = (
+  this: PluginContext,
+  chunk: RenderedChunk,
+) => string | Promise<string>
+
+export type AddonHook = string | AddonHookFunction
+
+export interface OutputPlugin
```

Comment:
The `OutputPlugin` typing is not used at rolldown, we could remove it at future if user not used.